spec/scenarios/credit_note_spec.rb

```diff
@@ -336,4 +336,547 @@
+          expect(invoice.total_amount_cents).to eq(163_99)
+          expect(invoice.taxes_amount_cents).to eq(27_33)
+          fees = invoice.fees
+          invoice.update(payment_status: 'succeeded')
+
+          # estimate and create credit notes for first item - full refund; the taxes are rounded to higher number
+          estimate_credit_note(
+            invoice_id: invoice.id,
+            items: [
+              {
+                fee_id: fees[0].id,
+                amount_cents: 68_33
+              }
+            ]
+          )
+
+          # Estimate the credit notes amount on one fee rounds the taxes to higher number
+          estimate = json[:estimated_credit_note]
+          expect(estimate).to include(
+            taxes_amount_cents: 13_67,
+            precise_taxes_amount_cents: "1366.6",
+            sub_total_excluding_taxes_amount_cents: 6833,
+            max_creditable_amount_cents: 8200,
+            max_refundable_amount_cents: 8200,
+            taxes_rate: 20.0
+          )
+
+          # Emit a credit note on only one fee
+          create_credit_note(
+            invoice_id: invoice.id,
+            reason: :other,
+            credit_amount_cents: 0,
+            refund_amount_cents: 82_00,
+            items: [
+              {
+                fee_id: fees[0].id,
+                amount_cents: 68_33
+              }
+            ]
+          )
+
+          credit_note = invoice.credit_notes.order(:created_at).last
+          expect(credit_note).to have_attributes(
+            refund_amount_cents: 82_00,
+            total_amount_cents: 82_00,
+            taxes_amount_cents: 13_67,
+            precise_taxes_amount_cents: 1366.6
+          )
+          expect(credit_note.precise_total).to eq(8199.6)
+          expect(credit_note.taxes_rounding_adjustment).to eq(0.4)
+
+          # when issuing second credit note, it should be rounded to lower number
+          estimate_credit_note(
+            invoice_id: invoice.id,
+            items: [
+              {
+                fee_id: fees[1].id,
+                amount_cents: 68_33
+              }
+            ]
+          )
+
+          estimate = json[:estimated_credit_note]
+          expect(estimate).to include(
+            taxes_amount_cents: 13_66,
+            precise_taxes_amount_cents: "1366.2",
+            sub_total_excluding_taxes_amount_cents: 6833,
+            max_creditable_amount_cents: 8199,
+            max_refundable_amount_cents: 8199,
+            taxes_rate: 20.0
+          )
+
+          # Emit a credit note on only one fee
+          create_credit_note(
+            invoice_id: invoice.id,
+            reason: :other,
+            credit_amount_cents: 0,
+            refund_amount_cents: 81_99,
+            items: [
+              {
+                fee_id: fees[1].id,
+                amount_cents: 68_33
+              }
+            ]
+          )
+
+          credit_note = invoice.credit_notes.order(:created_at).last
```

Comment:
I don't know a lot about credit notes but as I understand we create one credit note by invoice fee, then, the adjustment is done in the last one, what would happen if the last one is voided? I guess the sums still match but the voided credit not won't match the invoice fee... maybe I'm just mixing things.

Reply:
discussed it at slack, leaving here for the history:
I checked the algorithm behind voiding credit notes - we update their status and balance_amount, while the amounts do not change.
So if we void the last credit note, we won't be able to issue new credit_notes, but previous ones will keep their numbers, and that first CN for amount 81_99.6 will keep it's rounding to 82_00 , which is still mathematically correct.
the issue that we're solving here is the sum of all credit notes should not be bigger than invoice.total_amount, while technically on the level of each credit_note calculation even the current rounding is correct.